go/types/set_iterator.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package types
+
+import "github.com/attic-labs/noms/go/d"
+
+type SetIterator interface {
+	// Next returns nil, when no objects remain
+	Next() Value
+
+	// AdvanceTo(v) returns:
+	// * nil if there are no objects >= v.
+	// * the next value in the iteration if  v <= curValue
+	// * the first value > v if v is not in the set, or nil if there is no value > v
```

Comment:
The second part of this point "nil if there is no value > v" contradicts the first point "nil if there are no objects >= v".